### PR TITLE
Deprecate `skip_exit_code` in `BashOperator`

### PR DESCRIPTION
**Pull request by eladkal**
same as https://github.com/apache/airflow/pull/30733 just for core

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/operators/bash.py [hunk 1]
```diff
@@ -141,14 +142,18 @@ def __init__(
         append_env: bool = False,
         output_encoding: str = "utf-8",
         skip_exit_code: int = 99,
```
Comment (uranusjr) [inline airflow/operators/bash.py]: Wouldn’t this always trigger the deprecation warning? We need to change the default `skip_exit_code` to None.

File: airflow/operators/bash.py [hunk 2]
```diff
@@ -141,14 +142,18 @@ def __init__(
...
         cwd: str | None = None,
         **kwargs,
     ) -> None:
         super().__init__(**kwargs)
         self.bash_command = bash_command
         self.env = env
         self.output_encoding = output_encoding
-        self.skip_exit_code = skip_exit_code
+        if skip_exit_code:
+            warnings.warn("skip_exit_code is deprecated. Please use skip_on_exit_code")
```
Comment (uranusjr) [inline airflow/operators/bash.py]: Need `DeprecationWarning` (the default is `UserWarning`) and `stacklevel`

File: airflow/operators/bash.py (hunk 1 above)
Comment (hussein-awala) [reply airflow/operators/bash.py]: +1

File: airflow/operators/bash.py [hunk 3]
```diff
@@ -140,15 +141,20 @@ def __init__(
         env: dict[str, str] | None = None,
         append_env: bool = False,
         output_encoding: str = "utf-8",
-        skip_exit_code: int = 99,
+        skip_exit_code: int | None = None,
+        skip_on_exit_code: int = 99,
```
Comment (hussein-awala) [inline airflow/operators/bash.py:146]: I wonder if should set `None` as default value for `skip_on_exit_code` too, like by default the operator doesn't skip for any exit code.

Comment (eladkal) [reply airflow/operators/bash.py:146]: Changing default is a breaking change so we can't :(

**Review by hussein-awala — APPROVED** [review]

File: airflow/operators/bash.py [hunk 4]
```diff
@@ -140,15 +141,21 @@ def __init__(
...
         self.bash_command = bash_command
         self.env = env
         self.output_encoding = output_encoding
-        self.skip_exit_code = skip_exit_code
+        if skip_exit_code is not None:
+            warnings.warn(
+                "skip_exit_code is deprecated. Please use skip_on_exit_code", DeprecationWarning, stacklevel=2
+            )
+            self.skip_on_exit_code = skip_exit_code
+        self.skip_on_exit_code = skip_on_exit_code
```
Comment (hussein-awala) [inline airflow/operators/bash.py]: I think `self.skip_on_exit_code = skip_on_exit_code` override `self.skip_on_exit_code = skip_exit_code` and replace it by `None`, you need to put it in `else`

**Review by hussein-awala — CHANGES_REQUESTED** [review]

**Review by hussein-awala — APPROVED** [review]
looks good

**Comment by potiuk** [comment]
Should we also add the "Container" logic here to be fully consistent? I believe it should be fully backwards compatible ?

**Comment by potiuk** [comment]
Ah... you just merged it @eladkal  while I was writing the comment :)

**Comment by potiuk** [comment]
Here it is #30739

**Pull request merged**